cowbird/services/impl/geoserver.py

```diff
@@ -479,31 +494,37 @@ def _remove_shapefile_request(self, workspace_name, datastore_name, filename):
 @shared_task(bind=True, base=RequestTask)
 def create_workspace(self, user_name):
     # Avoid any actual logic in celery task handler, only task related stuff should be done here
-    return ServiceFactory().get_service("Geoserver").create_workspace(user_name)
+    return Geoserver.get_instance().create_workspace(user_name)
 
 
 @shared_task(bind=True, base=RequestTask)
 def create_datastore(self, datastore_name):
     # Avoid any actual logic in celery task handler, only task related stuff should be done here
-    return ServiceFactory().get_service("Geoserver").create_datastore(datastore_name)
+    return Geoserver.get_instance().create_datastore(datastore_name)
 
 
 @shared_task(bind=True, base=RequestTask)
 def remove_workspace(self, workspace_name):
     # Avoid any actual logic in celery task handler, only task related stuff should be done here
-    return ServiceFactory().get_service("Geoserver").remove_workspace(workspace_name)
+    return Geoserver.get_instance().remove_workspace(workspace_name)
 
 
 @shared_task(bind=True, autoretry_for=(FileNotFoundError,), retry_backoff=True, max_retries=8)
 def validate_shapefile(self, workspace_name, shapefile_name):
-    return ServiceFactory().get_service("Geoserver").validate_shapefile(workspace_name, shapefile_name)
+    return Geoserver.get_instance().validate_shapefile(workspace_name, shapefile_name)
 
 
 @shared_task(bind=True, base=RequestTask)
 def publish_shapefile(self, workspace_name, shapefile_name):
-    return ServiceFactory().get_service("Geoserver").publish_shapefile(workspace_name, shapefile_name)
+    return Geoserver.get_instance().publish_shapefile(workspace_name, shapefile_name)
 
 
 @shared_task(bind=True, base=RequestTask)
 def remove_shapefile(self, workspace_name, shapefile_name):
-    return ServiceFactory().get_service("Geoserver").remove_shapefile(workspace_name, shapefile_name)
+    return Geoserver.get_instance().remove_shapefile(workspace_name, shapefile_name)
+
+
+class GeoserverError(Exception):
+    """
+    Generic Geoserver error used to break request chains.
```

Comment:
Maybe just say why : `...because not covered by the ``RequestTask`` retry_for exceptions`